src/hotspot/share/services/threadService.cpp

```diff
@@ -1439,7 +1439,16 @@ oop ThreadSnapshotFactory::get_thread_snapshot(jobject jthread, TRAPS) {
 
   ResourceMark rm(THREAD);
   HandleMark   hm(THREAD);
-  Handle thread_h(THREAD, JNIHandles::resolve(jthread));
+
+  JavaThread* java_thread = nullptr;
+  oop thread_oop;
+  bool has_javathread = tlh.cv_internal_thread_to_JavaThread(jthread, &java_thread, &thread_oop);
+  Handle thread_h(THREAD, thread_oop);
+  bool is_virtual = java_lang_VirtualThread::is_instance(thread_h());
+
+  if (!has_javathread && !is_virtual) {
+    return nullptr; // thread terminated
```

Comment:
```suggestion
    return nullptr; // thread terminated so not of interest
```
It took me a lot of backtracking in the Java code to find ThreadContainer which finally stated that only live threads are of interest, so an expanded comment here would help me. ( I was ready to suggest we should be creating a `ThreadSnapshot` that represents the terminated thread.)

Reply:
Yes, although we could return _something_ for a terminated thread, that seems like a fiction that might do us no good, forgetting it and moving on seems good...